e2e/cypress/integration/plugins/plugin_buttons_spec.js

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+/**
+ * Note: This test requires test and demo plugin tar file under fixtures folder.
+ * Download from: https://github.com/mattermost/mattermost-plugin-test/releases/download/v0.1.0/com.mattermost.test-plugin-0.1.0.tar.gz
+ * Copy to: ./e2e/cypress/fixtures/com.mattermost.test-plugin-0.1.0.tar.gz
+ *
+ * Download version 0.1.0 from :
+ * https://github.com/mattermost/mattermost-plugin-demo/releases/download/v0.2.0/com.mattermost.demo-plugin-0.2.0.tar.gz
+ * Copy to : ./e2e/cypress/fixtures/com.mattermost.demo-plugin-0.2.0.tar.gz
+ */
+
+describe('collapse on 5 plugin buttons', () => {
+    let testTeam;
+
+    beforeEach(() => {
+        // # Set plugin settings
+        const newSettings = {
+            PluginSettings: {
+                Enable: true,
+                RequirePluginSignature: false,
+            },
+        };
+        cy.apiUpdateConfig(newSettings);
+
+        // # Login as Admin
+        cy.apiInitSetup().then(({team}) => {
+            testTeam = team;
+        });
+        cy.apiAdminLogin();
+    });
+
+    it('MM-T1649 Greater than 5 plugin buttons collapse to one icon in top nav', () => {
+        // # Go to town square
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        // # Try to remove test plugin, just in case
+        cy.apiRemovePluginById('com.mattermost.test-plugin');
+
+        // # Try to remove demo plugin, just in case
+        cy.apiRemovePluginById('com.mattermost.demo-plugin');
+
+        // # Upload and enable test plugin with 5 channel header icons
+        cy.apiUploadPlugin('com.mattermost.test-plugin-0.1.0.tar.gz').then(() => {
+            cy.apiEnablePluginById('com.mattermost.test-plugin');
+        });
+
+        // * Validate that there are 10 channel header icons
+        cy.get('.channel-header__icon').should('have.length', 10);
```

Comment:
It is not obvious that apart of the plugin icons, there are other 5 icons on the channel header. Also, it may become quite brittle if any of those icons is removed from the channel header. Is there any better way to check this?

Reply:
There are two ways we could improve this:
1. Add dynamically calculated id-s to the channel header buttons which is not trivial amount of work.
2. Do not validate exact number of icons like 10, but verify the difference of 4 when adding an extra icon.
I'm 0/5 if we should do either of them. Maybe @DHaussermann will have some insight

Reply:
Can we add a css class to the plugin buttons and count them that way?